apps/site/src/app/admin/participants/components/ParticipantAction.tsx

```diff
@@ -1,26 +1,68 @@
+import { useContext } from "react";
+
 import Button from "@cloudscape-design/components/button";
 
+import UserContext from "@/lib/admin/UserContext";
+import { isCheckinLead } from "@/lib/admin/authorization";
+import { Decision, PostAcceptedStatus } from "@/lib/admin/useApplicant";
 import { Participant } from "@/lib/admin/useParticipants";
+import ParticipantActionPopover from "./ParticipantActionPopover";
 
 interface ParticipantActionProps {
 	participant: Participant;
 	initiateCheckIn: (participant: Participant) => void;
+	initiatePromotion: (participant: Participant) => void;
 }
+
 function ParticipantAction({
 	participant,
 	initiateCheckIn,
+	initiatePromotion,
 }: ParticipantActionProps) {
-	// TODO: waitlist promotion
+	const { role } = useContext(UserContext);
+
+	const isCheckin = isCheckinLead(role);
+	const isWaiverSigned = participant.status === PostAcceptedStatus.signed;
+
+	const promoteButton = (
+		<Button
+			variant="inline-link"
+			ariaLabel={`Promote ${participant._id} off waitlist`}
+			onClick={() => initiatePromotion(participant)}
+			disabled={!isCheckin}
+		>
+			Promote
+		</Button>
+	);
 
-	return (
+	const checkinButton = (
 		<Button
 			variant="inline-link"
 			ariaLabel={`Check in ${participant._id}`}
 			onClick={() => initiateCheckIn(participant)}
+			disabled={isWaiverSigned}
 		>
 			Check In
 		</Button>
 	);
+
+	if (participant.status === Decision.waitlisted) {
+		if (!isCheckin) {
+			return (
+				<ParticipantActionPopover content="Only check-in leads are allowed to promote walk-ins.">
+					{promoteButton}
+				</ParticipantActionPopover>
+			);
+		}
+		return promoteButton;
+	} else if (isWaiverSigned) {
+		return (
+			<ParticipantActionPopover content="Must confirm attendance in portal first">
+				{checkinButton}
+			</ParticipantActionPopover>
+		);
+	}
+	return checkinButton;
```

Comment:
Issue: a participant with `ACCEPTED` status still needs to sign the waiver, so the check-in button would need to be disabled

Reply:
Button is still not disabled

Reply:
Sorry about that, should be disabled now.